### PR TITLE
Implement the state calculator component without Lit

### PR DESCRIPTION
**Pull request by oyamauchi**
## Description

We do not need Lit's update lifecycle management; we have React for
that. This is now a manually-implemented web component whose sole
purpose is to wrap the React component and tell it about updates to
the HTML attributes.

Bundle size is back down to 504K uncompressed, which is smaller than
before the commit where I added React(!) (552K, and that's without
the SVG icons bundled in).

## Test Plan

Grep for imports from `lit` to make sure they're only in files that
are not imported from the new calculator.

- Change the element's `lang` attribute, make sure the UI switches to 
  Spanish.

- Check the "include beta states" box, then enter the ZIP 12084 (NY)
  and make sure the utilities menu populates. Uncheck the box, delete
  and reenter the same ZIP, make sure the utilities menu doesn't
  populate (showing the checkbox takes effect immediately).

- Change one of the form-value attributes and make sure nothing changes
  immediately. Click "reset" and make sure the changed attribute value
  appears in the form.

- Delete all the CSS from `state-calculator.html` and make sure the
  calculator still has the proper styles (i.e. it wasn't accidentally
  relying on inheriting anything from the parent page).

### CODE REVIEW

File: src/state-calculator.tsx [hunk 1]
```diff
@@ -429,16 +304,160 @@ export const StateCalculator: FC<{
...
+  reactRootCalculator: Root;
+  reactRootFooter: Root;
+
+  constructor() {
+    super();
+
+    const shadowRoot = this.attachShadow({ mode: 'open' });
+
+    const style = document.createElement('style');
+    style.textContent = tailwindStyles;
```
Comment (oyamauchi) [inline src/state-calculator.tsx:359]: The old code includes `baseVariables` in the shadow styles, but that isn't necessary anymore.

File: src/styles.ts [hunk 2]
```diff
@@ -88,7 +88,7 @@ export const baseVariables = css`
     width: 100%;
     max-width: 1280px;
   }
-}`;
```
Comment (oyamauchi) [inline src/styles.ts:91]: This is a drive-by. Not necessary since the new code doesn't use `baseVariables` anymore, but I found it, so why not fix it

File: src/state-calculator.tsx [hunk 3]
```diff
@@ -429,16 +304,160 @@ export const StateCalculator: FC<{
...
+      this.render();
+    }
+  }
+
+  private render() {
+    this.reactRootCalculator.render(
+      <LocaleContext.Provider value={this.lang}>
+        <StateCalculator
+          shadowRoot={this.shadowRoot!}
+          language={this.lang}
```
Comment (RandomEtc) [inline src/state-calculator.tsx:420]: Do we need lang here if it's coming from a context?

Comment (oyamauchi) [reply src/state-calculator.tsx:420]: Yeah, because we need to pass it to the API.

Comment (oyamauchi) [reply src/state-calculator.tsx:420]: Although I suppose the API callsite could take it from React context directly?

File: src/state-calculator.tsx [hunk 4]
```diff
@@ -432,16 +298,159 @@ export const StateCalculator: FC<{
...
+  connectedCallback() {
+    this.render();
+  }
+
+  attributeChangedCallback(
+    attr: (typeof CalculatorElement.observedAttributes)[number],
+    _: string | null, // old value; unused
+    newValue: string | null,
+  ) {
+    if (attr === 'lang') {
```
Comment (RandomEtc) [inline src/state-calculator.tsx:375]: Does Typescript help us if we miss an attr here? Would a switch statement help?

Idle thought, not a blocker. Wall of `else if` or wall of `case` - the difference seems superficial if there's no tooling/linting benefit.

**Review by RandomEtc — APPROVED** [review]
Seems sound to me. I walked through the test plan in dev tools as well. Confirmed that all works in Safari as described.

File: src/state-calculator.tsx (hunk 4 above)
Comment (oyamauchi) [reply src/state-calculator.tsx:375]: Yes, see line 399. The type of `attr` is a big union of constants, and each `else if` case removes one constant from it. In the final `else`, it should be an empty union, i.e. `never`, and assigning it to a variable of type `never` asserts that.

A switch could do the same, but I did it this way to avoid the noise of `break;` in every case.
